third_party/move/move-compiler/src/parser/syntax.rs

```diff
@@ -947,8 +947,8 @@ fn parse_term(context: &mut Context) -> Result<Exp, Box<Diagnostic>> {
         },
         Tok::Identifier
             if context.tokens.content() == FOR_IDENT
-                && matches!(context.tokens.lookahead_nth(1), Ok(Tok::LParen))
-                && matches!(context.tokens.lookahead_nth(3), Ok(Tok::Identifier)) =>
+                && matches!(context.tokens.lookahead_nth(0), Ok(Tok::LParen))
+                && matches!(context.tokens.lookahead_nth(2), Ok(Tok::Identifier)) =>
```

Comment:
Yes, thank you very much for your reply, while querying the cause of this problem, I found that lookahead_nth was behaving strangely, so, but fixing it may cause problems elsewhere. 

 I can go ahead and fix this current_offset += length; and restore this pr